### PR TITLE
[TEST] Adding benchmarks for MSA.

### PR DESCRIPTION
**Pull request by simonsasse**
This PR adds benchmarks for the MSA. 
All combinations of long/short and few/many 
with both random and similar sequence sets.

I also added two sequence set generator functions.
This solves https://github.com/seqan/product_backlog/issues/186

### CODE REVIEW

File: test/include/seqan3/test/performance/sequence_generator.hpp [hunk 1]
```diff
@@ -91,6 +91,58 @@ auto generate_sequence_pairs(size_t const sequence_length,
...
+    std::uniform_int_distribution<> distrib_pos(0, sequence_length-1); // distribution for choosing a random position.
+    std::uniform_int_distribution<> distrib_base(0, 3); // distribution for choosing a random base.
+
+    std::vector<sequence_t> vec;
+    // generate first sequences
+    sequence_t seq = generate_sequence<alphabet_t>(sequence_length, 0, 1);
+    vec.push_back(seq);
+
+    size_t mutations = std::round(sequence_length * mutation_rate); // number of mutations per sequence.
+    for (size_t i = 0; i < set_size-1; ++i)
```
Comment (smehringer) [inline test/include/seqan3/test/performance/sequence_generator.hpp]:
```suggestion
    for (size_t i = 0; i < set_size - 1; ++i)
```
:nail_care:

File: test/include/seqan3/test/performance/sequence_generator.hpp [hunk 2]
```diff
@@ -91,6 +91,58 @@ auto generate_sequence_pairs(size_t const sequence_length,
...
+    {
+        // mutate the original sequence.
+        sequence_t seq_temp = seq;
+        for ( size_t i=0 ; i < mutations ; i++ )
```
Comment (smehringer) [inline test/include/seqan3/test/performance/sequence_generator.hpp]:
```suggestion
        for ( size_t i = 0 ; i < mutations ; i++ )
```

File: test/include/seqan3/test/performance/sequence_generator.hpp [hunk 3]
```diff
@@ -91,6 +91,58 @@ auto generate_sequence_pairs(size_t const sequence_length,
...
+template <typename alphabet_t>
+auto generate_similar_sequence_set(size_t const sequence_length,
+                                   size_t const set_size,
+                                   float const mutation_rate)
+{
+    using sequence_t = decltype(generate_sequence<alphabet_t>());
+
+    std::random_device rd;
+    std::mt19937 gen(rd());
+    std::uniform_int_distribution<> distrib_pos(0, sequence_length-1); // distribution for choosing a random position.
```
Comment (smehringer) [inline test/include/seqan3/test/performance/sequence_generator.hpp]:
```suggestion
    std::uniform_int_distribution<> distrib_pos(0, sequence_length - 1); // distribution for choosing a random position.
```

File: test/include/seqan3/test/performance/sequence_generator.hpp [hunk 4]
```diff
@@ -91,6 +91,58 @@ auto generate_sequence_pairs(size_t const sequence_length,
...
+
+    size_t mutations = std::round(sequence_length * mutation_rate); // number of mutations per sequence.
+    for (size_t i = 0; i < set_size-1; ++i)
+    {
+        // mutate the original sequence.
+        sequence_t seq_temp = seq;
+        for ( size_t i=0 ; i < mutations ; i++ )
+        {
+            size_t pos = distrib_pos(gen);
+            seq_temp[pos]=seq_temp[pos].assign_rank(distrib_base(gen));
```
Comment (smehringer) [inline test/include/seqan3/test/performance/sequence_generator.hpp]:
```suggestion
            seq_temp[pos] = seq_temp[pos].assign_rank(distrib_base(gen));
```

File: test/performance/alignment/multiple_alignment_benchmark.cpp [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+}
+
+static void seqan3_msa_random_sequences(benchmark::State & state)
+{
+    auto n = static_cast<size_t>(state.range(0));
+    assert(n > 1);
+    auto i = static_cast<size_t>(state.range(1));
+    assert(i > 0);
+    // generate random sequence set.
+    auto seqs = seqan3::test::generate_random_sequence_set<seqan3::dna4>(i, n, 1);
```
Comment (smehringer) [inline test/performance/alignment/multiple_alignment_benchmark.cpp]: Can you tell me why you chose `sequence_variance = 1` and what the implications are?

**Review by smehringer — CHANGES_REQUESTED** [review]
Looks great, good job!

Only some :nail_care: and a question

One thing for now: Can you move the `generate...set` function into the benchmark.cpp file for now?
Lydia is currently working to refactor the generate_sequencing stuff and this way we stay independent of that and can adjust it later.

File: test/performance/alignment/multiple_alignment_benchmark.cpp (hunk 5 above)
Comment (simonsasse) [reply test/performance/alignment/multiple_alignment_benchmark.cpp]: No actually not. I think this is a relict from me playing around with these values. I should have reset it to zero as is the default of all the other seq generator function. I think the randomness of the bases is created by varying the seed. Varying the sequence variance makes the resulting sequences vary in length. I think we don't need that plus the gerate_similar_seqeunce_set() does not do it either. 
I'll set it back to zero 👍

**Review by smehringer — APPROVED** [review]

**Comment by smehringer** [comment]
I won't wait for CI since it doesn't test seqan2 stuff anyway

**Pull request merged**